data.py

```diff
@@ -0,0 +1,76 @@
+"""Parsing and data-related functions
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import can_pb2
+import validator
+
+from google.protobuf import text_format
+
+import re
+from constants import NUM_CAN_DEVICES, NUM_CAN_MESSAGES
+
+
+def read_protobuf_data(filename):
+    """Reads and parses the specified ASCII Pb
+
+    Args:
+        filename: a string representing the filename to read
+
+    Returns:
+        a list of Pb values
+    """
+    can_messages = can_pb2.CanSchema()
+    with open(filename, 'r') as asciipb:
+        text_format.Merge(asciipb.read(), can_messages)
+    return can_messages.msg
+
+
+def parse_can_device_enum():
+    """Read and parse the CanMsg source devices
+
+    Args:
+        None
+
+    Returns:
+        a dictionary of CAN source devices and their names
+    """
+    can_devices = {}
+    for key in range(NUM_CAN_DEVICES):
+        can_devices[key] = None
+    for name, value in can_pb2.CanMsg.Source.items():
+        can_devices[value] = name
+    return can_devices
+
+
+def parse_can_message_enum():
+    """Parses CAN messages into dictionary
+    """
+    messages = {}
+    for i in range(128):
```

Comment:
This 128 should be in constants.py I think. 

Also, it is too bad the for compatibility between 2.7 and 3 we can't use xrange throughout as it is a lot faster.